src/mongocxx/config/prelude.hpp

```diff
@@ -12,7 +12,44 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// src/mongocxx/config/config.hpp.in
+#pragma push_macro("MONGOCXX_INLINE_NAMESPACE_BEGIN")
+#undef MONGOCXX_INLINE_NAMESPACE_BEGIN
+#pragma push_macro("MONGOCXX_INLINE_NAMESPACE_END")
+#undef MONGOCXX_INLINE_NAMESPACE_END
+
+// src/mongocxx/config/version.hpp.in
+#pragma push_macro("MONGOCXX_VERSION_EXTRA")
+#undef MONGOCXX_VERSION_EXTRA
+#pragma push_macro("MONGOCXX_VERSION_MAJOR")
+#undef MONGOCXX_VERSION_MAJOR
+#pragma push_macro("MONGOCXX_VERSION_MINOR")
+#undef MONGOCXX_VERSION_MINOR
+#pragma push_macro("MONGOCXX_VERSION_PATCH")
+#undef MONGOCXX_VERSION_PATCH
+
+// export.hpp (generated by cmake)
+#pragma push_macro("DEFINE_NO_DEPRECATED")
+#undef DEFINE_NO_DEPRECATED
+#pragma push_macro("MONGOCXX_API")
+#undef MONGOCXX_API
+#pragma push_macro("MONGOCXX_API_H")
+#undef MONGOCXX_API_H
+#pragma push_macro("MONGOCXX_DEPRECATED")
+#undef MONGOCXX_DEPRECATED
+#pragma push_macro("MONGOCXX_DEPRECATED_EXPORT")
+#undef MONGOCXX_DEPRECATED_EXPORT
+#pragma push_macro("MONGOCXX_DEPRECATED_NO_EXPORT")
+#undef MONGOCXX_DEPRECATED_NO_EXPORT
+#pragma push_macro("MONGOCXX_NO_DEPRECATED")
+#undef MONGOCXX_NO_DEPRECATED
+#pragma push_macro("MONGOCXX_PRIVATE")
+#undef MONGOCXX_PRIVATE
+
 #include <mongocxx/config/config.hpp>
+#include <mongocxx/config/version.hpp>
 #include <mongocxx/export.hpp>
 
+#pragma push_macro("MONGOCXX_INLINE")
+#undef MONGOCXX_INLINE
```

Comment:
Like in the bson side, this shouldn't be defined here. Add a TODO here as well, can be the same ticket as I suggested in the bson side. Ideally, MONGOCXX_INLINE would just expand to BSONCXX_INLINE, but we probably want a mongocxx/compiler.hpp file as well. It can just include the bson lib one.